src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_SIZE_ARRAY = 10;
+    private T[] myList;
+    private int size;
+
+    public ArrayList() {
+        myList = (T[]) new Object[MAX_SIZE_ARRAY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size >= myList.length) {
+            myList = expandArray();
+        }
+        myList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("This index is incorrect");
+        }
+        if (size == myList.length) {
+            myList = expandArray();
+        }
+        System.arraycopy(myList, index, myList, index + 1, size++ - index);
+        myList[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return myList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        myList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index > size || index < 0) {
```

Comment:
should you be able to remove from index 5 when your size is 5?

Reply:
yes. We must use checkIndex instead if